app/controllers/application_controller.rb

```diff
@@ -3,34 +3,17 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
-  helper_method :current_user
-  helper_method :user_signed_in?
-  helper_method :correct_user?
+  before_action :configure_devise_permitted_parameters, if: :devise_controller?
 
-  private
-    def current_user
-      begin
-        @current_user ||= User.find(session[:user_id]) if session[:user_id]
-      rescue Exception => e
-        nil
-      end
-    end
-
-    def user_signed_in?
-      return true if current_user
-    end
-
-    def correct_user?
-      @user = User.find(params[:id])
-      unless current_user == @user
-        redirect_to root_url, :alert => "Access denied."
-      end
-    end
+  protected
+  def configure_devise_permitted_parameters
+    registration_params = [:email, :password, :password_confirmation, :roles]
 
-    def authenticate_user!
-      if !current_user
```

Comment:
Porqué hay un métodos custom, devise debería hacer esto por tí si está correctamente configurado. Revisa la sección de "Controller filters and helpers" en https://github.com/plataformatec/devise.